source/scenes/gameplay/PPCanvasBlock.cpp

```diff
@@ -4,39 +4,47 @@ ptr<CanvasBlock> CanvasBlock::alloc(
     const asset_t &assets,
     float size,
     const vec<Color4>& colors, const int numCanvasColors,
-    const GameStateController &state) {
+    const GameStateController &state, bool isObstacle) {
     auto result = make_shared<CanvasBlock>();
     if (result->initWithBounds(Rect(0, 0, size, size)))
-        result->_setup(assets, colors, numCanvasColors, state);
+        result->_setup(assets, colors, numCanvasColors, state, isObstacle);
     else
         return nullptr;
     return result;
 }
 
 
-void CanvasBlock::_setup(const asset_t &assets, const vec<Color4>& colors, const int numCanvasColors, const GameStateController &state) {
+void CanvasBlock::_setup(const asset_t &assets, const vec<Color4>& colors, const int numCanvasColors, const GameStateController &state, bool isObstacle) {
 #ifdef VIEW_DEBUG
     auto n = PolygonNode::to(Rect(Vec2::ZERO, getContentSize()));
     n->setColor(Color4f(0, 1, 0, .3));
     addChild(n);
 #endif
+    _isObstacle = isObstacle; 
     _state = state; 
     _isActive = false;
     _initialColorNumber = numCanvasColors;
     /*string characters[] = {"panda", "bird-1", "bird-2", "cat-1", "cat-2",
                            "dog-1", "dog-2", "dog-3", "frog", "octopus"};*/
     string characters[] = { "husky", "samoyed", "cat1", "cat2"};
+    //Temporary fix until animation is complete
     int p = Random::getInstance()->getInt(4 - 1);
-    _texture_array[0] = assets->get<Texture>(characters[p] + "-blink");
-    _texture_array[1] = assets->get<Texture>(characters[p] + "-emotion-1");
-    _texture_array[2] = assets->get<Texture>(characters[p] + "-emotion-2");
-    _texture_array[3] = assets->get<Texture>(characters[p] + "-emotion-3");
+    if (isObstacle) {
+        _texture_array[0] = assets->get<Texture>("obstacle-inactive");
+        _texture_array[1] = assets->get<Texture>("obstacle-active");
+        _texture_array[2] = assets->get<Texture>("obstacle-explode");
+    }
```

Comment:
Lindsey is planning on having dif obstacles for dif worlds, so we'll have to rework this later, but this is good for now

Reply:
For sure. Hopefully that isn't a terrible fix to do